library/psa_crypto.c

```diff
@@ -324,11 +325,7 @@ psa_status_t mbedtls_to_psa_error( int ret )
 /* Key management */
 /****************************************************************/
 
-#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR) || \
-    defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_PUBLIC_KEY) || \
-    defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
-    defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) || \
-    defined(MBEDTLS_PSA_BUILTIN_ALG_ECDH)
+#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR)
 mbedtls_ecp_group_id mbedtls_ecc_group_of_psa( psa_ecc_family_t curve,
```

Comment:
(continuing https://github.com/ARMmbed/mbedtls/pull/5139#discussion_r816289534, https://github.com/ARMmbed/mbedtls/pull/5139#discussion_r777546822, https://github.com/ARMmbed/mbedtls/pull/5139#issuecomment-1057821528, but you don't need to read those to follow this discussion)

If neither `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR` nor `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_PUBLIC_KEY` is defined, this indicates that the built-in crypto does not support any elliptic curve key, and so `mbedtls_ecc_group_of_psa` can only ever return 0.  A configuration in which, for example, neither `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR` nor `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_PUBLIC_KEY` are defined, but `MBEDTLS_PSA_BUILTIN_ALG_ECDSA` is defined, is not useful: ECDSA can only be used with ECC keys, and you can't ever perform ECDSA if you can't create an ECC key.

Hence `defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR) || defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_PUBLIC_KEY)` is the right condition here. (Earlier I stated that this simplifies to `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_PUBLIC_KEY`, since you can't have private key support without public key support; but this is true for `WANT` symbols, but not necessarily for `BUILTIN` symbols since it's in principle possible to have an accelerator for ECC public key operations only, and thus the builtin ECC code would only be used for private operations.)

However, there are preexisting issues with the test scripts for which I do not have a simple solution. So I propose to keep the original condition for `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR`. Please reference https://github.com/ARMmbed/mbedtls/issues/5596 comment in the commit message.